client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -721,9 +786,10 @@ public Collection<String> getAllSubjectsByPrefix(String subjectPrefix) throws IO
 
   @Override
   public synchronized void reset() {
-    schemaCache.clear();
-    idCache.clear();
-    versionCache.clear();
+    schemaToIdCache.clear();
+    idToSchemaCache.clear();
+    schemaToVersionCache.clear();
+    versionToSchemaCache.clear();
```

Comment:
should we also `missingVersionCache.invalidateAll()`?

Reply:
Good catch, I'll add